x-pack/plugins/apm/server/plugin.ts

```diff
@@ -49,30 +57,23 @@ export class APMPlugin implements Plugin<APMPluginContract> {
       usageCollection?: UsageCollectionSetup;
     }
   ) {
-    const logger = this.initContext.logger.get('apm');
+    const logger = (this.logger = this.initContext.logger.get('apm'));
     const config$ = this.initContext.config.create<APMXPackConfig>();
     const mergedConfig$ = combineLatest(plugins.apm_oss.config$, config$).pipe(
       map(([apmOssConfig, apmConfig]) => mergeConfigs(apmOssConfig, apmConfig))
     );
 
     this.legacySetup$.subscribe(__LEGACY => {
-      createApmApi().init(core, { config$: mergedConfig$, logger, __LEGACY });
+      createApmApi().init(core, {
+        config$: mergedConfig$,
+        logger,
+        __LEGACY
+      });
     });
 
-    const currentConfig = await mergedConfig$.pipe(take(1)).toPromise();
-
-    // create agent configuration index without blocking setup lifecycle
-    createApmAgentConfigurationIndex({
-      esClient: core.elasticsearch.dataClient,
-      config: currentConfig,
-      logger
-    });
-    // create custom action index without blocking setup lifecycle
-    createApmCustomLinkIndex({
-      esClient: core.elasticsearch.dataClient,
-      config: currentConfig,
-      logger
-    });
+    const currentConfig = (this.currentConfig = await mergedConfig$
+      .pipe(take(1))
+      .toPromise());
```

Comment:
For simplicity let's stick to a single assignment

```suggestion
    this.currentConfig = await mergedConfig$
      .pipe(take(1))
      .toPromise();
```

Reply:
If using only the class variable, typescript is unable to infer if the class variable is defined in async context requiring a `!` to override the possibly undefined values. So I preferred the type safety of another local variable. I've addressed your suggestions in 4898cfccafeaab8f742ffa2c802d4be544c8da36 let me know what you think?